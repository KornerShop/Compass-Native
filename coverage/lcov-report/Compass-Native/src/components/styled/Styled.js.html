<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Compass-Native/src/components/styled/Styled.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">Compass-Native/src/components/styled</a> Styled.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">34% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>17/50</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">34% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>17/50</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from 'styled-components/native';
&nbsp;
export const styleSwitch = <span class="fstat-no" title="function not covered" >(p</span>rop, cssIfValid, cssIfInvalid) =&gt; {
<span class="cstat-no" title="statement not covered" >  switch (prop) {</span>
    case null:
<span class="cstat-no" title="statement not covered" >      return cssIfValid;</span>
    case true:
<span class="cstat-no" title="statement not covered" >      return cssIfValid;</span>
    case false:
<span class="cstat-no" title="statement not covered" >      return cssIfInvalid;</span>
    default:
<span class="cstat-no" title="statement not covered" >      return '';</span>
  }
};
&nbsp;
export const textInputColor = <span class="fstat-no" title="function not covered" >pr</span>op =&gt; {
<span class="cstat-no" title="statement not covered" >  switch (prop) {</span>
    case null:
<span class="cstat-no" title="statement not covered" >      return '#7C7A7A';</span>
    case true:
<span class="cstat-no" title="statement not covered" >      return '#7C7A7A';</span>
    case false:
<span class="cstat-no" title="statement not covered" >      return 'tomato';</span>
    default:
<span class="cstat-no" title="statement not covered" >      return '';</span>
  }
};
&nbsp;
export const textInputColorZip = <span class="fstat-no" title="function not covered" >pr</span>op =&gt; {
<span class="cstat-no" title="statement not covered" >  switch (prop) {</span>
    case null:
<span class="cstat-no" title="statement not covered" >      return 'white';</span>
    case true:
<span class="cstat-no" title="statement not covered" >      return 'white';</span>
    case false:
<span class="cstat-no" title="statement not covered" >      return 'tomato';</span>
    default:
<span class="cstat-no" title="statement not covered" >      return '';</span>
  }
};
&nbsp;
export const textInputWrapperColor = <span class="fstat-no" title="function not covered" >pr</span>op =&gt; {
<span class="cstat-no" title="statement not covered" >  switch (prop) {</span>
    case null:
<span class="cstat-no" title="statement not covered" >      return '1px solid papayawhip';</span>
    case true:
<span class="cstat-no" title="statement not covered" >      return '1px solid papayawhip';</span>
    case false:
<span class="cstat-no" title="statement not covered" >      return '1px solid tomato';</span>
    default:
<span class="cstat-no" title="statement not covered" >      return '';</span>
  }
};
&nbsp;
export const textInputWrapperColorZip = <span class="fstat-no" title="function not covered" >pr</span>op =&gt; {
<span class="cstat-no" title="statement not covered" >  switch (prop) {</span>
    case null:
<span class="cstat-no" title="statement not covered" >      return '2px solid white';</span>
    case true:
<span class="cstat-no" title="statement not covered" >      return '2px solid white';</span>
    case false:
<span class="cstat-no" title="statement not covered" >      return '2px solid tomato';</span>
    default:
<span class="cstat-no" title="statement not covered" >      return '';</span>
  }
};
&nbsp;
export const StyledContainer = styled.View`
  flex: 1;
  margin-top: 15;
  background-color: white;
  padding-top: 20;
  padding-left: 20;
  padding-right: 20;
`;
&nbsp;
export const ImageContainer = styled.Image`
  flex: 1;
  justify-content: center;
  align-items: center;
  width: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.width}</span>;
  height: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.height}</span>;
`;
&nbsp;
export const OfficeText = styled.Text`
  font-size: 50;
  color: white;
  background-color: transparent;
`;
&nbsp;
export const StyledInput = styled.TextInput`
  font-size: 15;
  font-weight: 300;
  color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >textInputColor(props.valid)}</span>;
  height: 30;
  border-color: papayawhip;
`;
&nbsp;
export const StyledInputZip = StyledInput.extend`
  font-size: 20;
  font-weight: 600;
  color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >textInputColorZip(props.valid)}</span>;
`;
&nbsp;
export const InputWrapper = styled.View`
  border: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >textInputWrapperColor(props.valid)}</span>;
  border-radius: 5;
  padding-top: 10;
  padding-bottom: 10;
  padding-left: 10;
  padding-right: 10;
  background-color: papayawhip;
`;
&nbsp;
export const ZipModalInputWrapper = InputWrapper.extend`
  border: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >textInputWrapperColorZip(props.valid)}</span>;
  border-top-width: 0;
  border-left-width: 0;
  border-right-width: 0;
  margin-left: 10;
  margin-right: 10;
  background-color: transparent;
`;
&nbsp;
export const LifeEventText = styled.Text`
  text-align: center;
  font-size: 18;
  font-weight: 300;
`;
&nbsp;
export const FormHeader = styled.Text`
  font-weight: bold;
  font-size: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; (<span class="cstat-no" title="statement not covered" >props.zip ? 35 : 20)</span>};
  text-align: center;
  padding-bottom: 10;
  color: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; (<span class="cstat-no" title="statement not covered" >props.zip ? 'white' : '#2C2C2C')</span>};
`;
&nbsp;
export const ActivityIndicatorWrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  background-color: #2c2c2c;
`;
&nbsp;
export const WelcomeUIWrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-content: center;
`;
&nbsp;
export const Logo = styled.Text`
  text-align: center;
  font-size: 65;
  font-weight: 300;
  color: white;
  background-color: transparent;
  margin-bottom: 30;
  font-family: merriweather-sans;
`;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jul 17 2017 07:56:03 GMT-0700 (PDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
