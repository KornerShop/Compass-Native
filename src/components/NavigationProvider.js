import React from 'react'
import {
  NavigationProvider,
  TabNavigation,
  StackNavigation,
} from '@expo/ex-navigation'
import Router from '../navigation/Router'

export default () =>
  <NavigationProvider router={Router}>
    <StackNavigation initialRoute={Router.getRoute('rootNavigation')} />
  </NavigationProvider>
