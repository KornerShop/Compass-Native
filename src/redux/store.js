import {createStore, compose, applyMiddleware} from 'redux'
import thunk from 'redux-thunk'
import rootReducer from './reducers'

export default createStore(rootReducer, compose(applyMiddleware(thunk)))
