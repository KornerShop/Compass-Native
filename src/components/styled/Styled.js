import styled from 'styled-components/native';

export const styleSwitch = (prop, cssIfValid, cssIfInvalid) => {
  switch (prop) {
    case null:
      return cssIfValid;
    case true:
      return cssIfValid;
    case false:
      return cssIfInvalid;
    default:
      return '';
  }
};

export const textInputColor = prop => {
  switch (prop) {
    case null:
      return '#7C7A7A';
    case true:
      return '#7C7A7A';
    case false:
      return 'tomato';
    default:
      return '';
  }
};

export const textInputColorZip = prop => {
  switch (prop) {
    case null:
      return 'white';
    case true:
      return 'white';
    case false:
      return 'tomato';
    default:
      return '';
  }
};

export const textInputWrapperColor = prop => {
  switch (prop) {
    case null:
      return '1px solid papayawhip';
    case true:
      return '1px solid papayawhip';
    case false:
      return '1px solid tomato';
    default:
      return '';
  }
};

export const textInputWrapperColorZip = prop => {
  switch (prop) {
    case null:
      return '2px solid white';
    case true:
      return '2px solid white';
    case false:
      return '2px solid tomato';
    default:
      return '';
  }
};

export const StyledContainer = styled.View`
  flex: 1;
  margin-top: 15;
  background-color: white;
  padding-top: 20;
  padding-left: 20;
  padding-right: 20;
`;

export const ImageContainer = styled.Image`
  flex: 1;
  justify-content: center;
  align-items: center;
  width: ${props => props.width};
  height: ${props => props.height};
`;

export const OfficeText = styled.Text`
  font-size: 50;
  color: white;
  background-color: transparent;
`;

export const StyledInput = styled.TextInput`
  font-size: 15;
  font-weight: 300;
  color: ${props => textInputColor(props.valid)};
  height: 30;
  border-color: papayawhip;
`;

export const StyledInputZip = StyledInput.extend`
  font-size: 20;
  width: 300;
  font-weight: 600;
  color: ${props => textInputColorZip(props.valid)};
`;

export const InputWrapper = styled.View`
  border: ${props => textInputWrapperColor(props.valid)};
  border-radius: 5;
  padding-top: 10;
  padding-bottom: 10;
  padding-left: 10;
  padding-right: 10;
  background-color: papayawhip;
`;

export const ZipModalInputWrapper = InputWrapper.extend`
  border: ${props => textInputWrapperColorZip(props.valid)};
  border-top-width: 0;
  border-left-width: 0;
  border-right-width: 0;
  margin-left: 10;
  margin-right: 10;
  background-color: transparent;
`;

export const LifeEventText = styled.Text`
  text-align: center;
  font-size: 18;
  font-weight: 300;
`;

export const FormHeader = styled.Text`
  font-weight: bold;
  font-size: ${props => (props.zip ? 35 : 20)};
  text-align: center;
  padding-bottom: ${props => (props.zip ? 80 : 10)};
  color: ${props => (props.zip ? 'white' : '#2C2C2C')};
`;

export const EligibleHeader = FormHeader.extend`
  font-size: 30;
  letter-spacing: 2;
  padding-top: 2;
  padding-right:10;
  padding-bottom:0;
  padding-left: 10;
`

export const ActivityIndicatorWrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  background-color: white;
`;

export const WelcomeUIWrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-content: center;
`;

export const Logo = styled.Text`
  text-align: center;
  font-size: 65;
  font-weight: 300;
  color: white;
  background-color: transparent;
  margin-bottom: 30;
  font-family: merriweather-sans;
`;
