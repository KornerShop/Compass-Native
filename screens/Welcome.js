import React, {Component} from 'react'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {View, Text, StyleSheet} from 'react-native'
import {ButtonGroup} from 'react-native-elements'

import {setLanguagePreference} from '../redux/actions/actions'

class Welcome extends Component {
  constructor(props) {
    super(props)
  }
  // Updates language preference, triggering the tabbed UI to render
  updateIndex(idx) {
    // Dispatch action to change the language key on our redux store,
    // which would hopefully update the 'selected' on line 26
    // if it doesn't, let's talk
    const language = idx === 1 ? 'es' : 'en'
    this.props.setLanguagePreference(language)
    this.props.toggleStart()
  }
  render() {
    return (
      <ButtonGroup
        onPress={this.updateIndex.bind(this)}
        selectedIndex={this.props.language === 'en' ? 0 : 1}
        buttons={['English', 'EspaÃ±ol']}
        containerStyle={{height: 150}}
      />
    )
  }
}

const styles = StyleSheet.create({})

const mapStateToProps = ({language}) => {
  language
}

const mapDispatchToProps = dispatch => {
  setLanguagePreference: bindActionCreators(setLanguagePreference, dispatch)
}

export default connect()(Welcome)
