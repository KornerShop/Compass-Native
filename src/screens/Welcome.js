import React, {Component} from 'react'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {
  Image,
  View,
  Text,
  StyleSheet,
  AsyncStorage,
  Dimensions,
  ActivityIndicator,
  StatusBar,
} from 'react-native'
import {FontAwesome} from '@expo/vector-icons'
import {ButtonGroup} from 'react-native-elements'
import styled from 'styled-components/native'

import {setLanguagePreference} from '../redux/actions/actions'

// update office state
// custom marker views
// Modal: update location state; update office state
// translations
// submit button on form
// lock in form style
// eligible and ineligible screens
// test on Android
// uncomment out get AsyncStorage in Main.js
// take stock of issues, re-organize via waffle
// organize code
// tests
// look into reprecussions of ejecting w/ regard to pushing to app stores
// smile - we can do it!

const ActivityIndicatorWrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`

const BackgroundImage = props =>
  <Image>
    {props.children}
  </Image>

const WelcomeUIContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-content: center;
`

const Logo = styled.Text`
  text-align: center;
  font-size: 65;
  font-weight: 300;
  color: white;
  background-color: transparent;
  margin-bottom: 30
`

class Welcome extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLoading: true,
      selectedLanguage: 2,
    }
  }

  async updateIndex(idx) {
    this.setState({selectedLanguage: idx})
    const language = idx === 1 ? 'es' : 'en'
    try {
      await AsyncStorage.setItem('language', language)
    } catch (error) {
      console.warn(error)
    }
    this.props.setLanguagePreference(language)
    this.props.toggleStart()
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <StatusBar barStyle="light-content" />
        <Image
          onLoad={() => this.setState({isLoading: false})}
          source={require('../assets/shopper.png')}
          tintColor={'red'}
          style={{
            flex: 1,
            height: this.props.orientation.height,
            width: this.props.orientation.width,
            resizeMode: 'cover',
          }}>
          {this.state.isLoading
            ? <ActivityIndicatorWrapper>
                <ActivityIndicator color="tomato" size="large" />
              </ActivityIndicatorWrapper>
            : <WelcomeUIContainer>
                <Logo>
                  C<FontAwesome name="compass" size={50} color="white" />mpass
                </Logo>
                <ButtonGroup
                  onPress={this.updateIndex.bind(this)}
                  selectedIndex={this.state.selectedLanguage}
                  buttons={['English', 'EspaÃ±ol']}
                  textStyle={{color: 'white', fontSize: 18, fontWeight: 'bold'}}
                  selectedTextStyle={{color: 'slategray'}}
                  innerBorderStyle={{
                    color: 'white',
                    width: 3,
                  }}
                  containerStyle={{
                    height: 60,
                    marginTop: 30,
                    backgroundColor: 'transparent',
                    borderWidth: 3,
                    borderRadius: 7,
                    borderColor: 'white',
                    width: 300,
                    alignSelf: 'center',
                  }}
                  selectedBackgroundColor="white"
                />
              </WelcomeUIContainer>}
        </Image>
      </View>
    )
  }
}

const mapStateToProps = ({orientation}) => ({orientation})

const mapDispatchToProps = dispatch => ({
  setLanguagePreference: bindActionCreators(setLanguagePreference, dispatch),
})

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
