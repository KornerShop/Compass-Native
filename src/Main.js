import React from 'react'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'

import {
  AsyncStorage,
  ActivityIndicator,
  View,
  StyleSheet,
  Dimensions,
} from 'react-native'

import NavigationProvider from './components/NavigationProvider'
import Welcome from './screens/Welcome'

import {updateOrientation} from './redux/actions/actions'
import {setLanguagePreference} from './redux/actions/actions'

class Main extends React.Component {
  constructor() {
    super()
    this.state = {
      started: false,
      isLoading: true,
    }
  }
  toggleStart() {
    this.setState({started: !this.state.started})
  }
  async componentWillMount() {
    // Not ideal, but this action is necessary for getting the initial orientation of the device
    this.props.updateOrientation(Dimensions.get('window'))
    try {
      const language = await AsyncStorage.getItem('language')
      this.setState({
        isLoading: false,
      })
      if (language) {
        this.setState({
          started: true,
        })
        setLanguagePreference(language)
      }
    } catch (error) {
      console.warn(error)
    }
  }
  renderRoot(Component) {
    return (
      <View
        style={{flex: 1}}
        onLayout={() => this.props.updateOrientation(Dimensions.get('window'))}>
        <Component toggleStart={this.toggleStart.bind(this)} />
      </View>
    )
  }
  render() {
    if (this.state.isLoading) {
      return (
        <View style={styles.loader}>
          <ActivityIndicator color="tomato" size="large" />
        </View>
      )
    }
    return this.state.started
      ? this.renderRoot(NavigationProvider)
      : this.renderRoot(Welcome)
  }
}

const mapDispatchToProps = dispatch => ({
  updateOrientation: bindActionCreators(updateOrientation, dispatch),
  setLanguagePreference: bindActionCreators(setLanguagePreference, dispatch),
})

const styles = StyleSheet.create({
  loader: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'papayawhip',
  },
})

export default connect(null, mapDispatchToProps)(Main)
