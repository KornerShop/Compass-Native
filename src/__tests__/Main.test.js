import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';
import { Provider } from 'react-redux';

import ActivityIndicator from 'react-native';

import Main, { Unwrapped as UnwrappedMain } from '../Main';
import Welcome from '../screens/Welcome';
import { ActivityIndicatorWrapper } from '../components/styled/Styled';
import NavigationProvider from '../containers/NavigationProvider';

import store from '../redux/store';

const wait = (exp, ms) =>
  new Promise(resolve => setTimeout(() => resolve(exp), ms));

describe('shallow tests', () => {
  test('Main renders correctly', () => {
    const wrapper = shallow(<UnwrappedMain />);
    expect(wrapper.state()).toMatchObject({
      started: false,
      isLoading: true,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('Main renders Welcome after initial load', async () => {
    const wrapper = shallow(<UnwrappedMain />);
    await wait(wrapper.setState({ isLoading: false }), 100);
    expect(wrapper.find(Welcome)).toBeTruthy();
  });

  test('Main renders navigationProvider after language chosen', async () => {
    const wrapper = shallow(<UnwrappedMain />);
    await wait(wrapper.setState({ isLoading: false }), 100);
    wrapper.setState({ started: true });
    expect(wrapper.find(NavigationProvider).toBeTruthy);
  });
})

// can you test Welcome's 'on tap' event to render NavigationProvider?

// Full DOM Rendering https://github.com/airbnb/enzyme
// Static Rendering https://github.com/airbnb/enzyme
