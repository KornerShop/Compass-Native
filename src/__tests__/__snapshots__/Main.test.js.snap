// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<Styled(View)>
  <ActivityIndicator
    animating={true}
    color="#00897b"
    hidesWhenStopped={true}
    size="large"
  />
</Styled(View)>
`;
